bridge-history-api/internal/controller/fetcher/l1_fetcher.go

```diff
@@ -153,6 +153,7 @@ func (c *L1MessageFetcher) fetchAndSaveEvents(confirmation uint64) {
 			return
 		}
 
+		c.l1FetcherLogic.ShiftToNextLastFinalizedBatchIndex()
```

Comment:
I think updating the status here is enough. reorg won't affect this info. pls revisit this logic a bit. @amoylan2

Reply:
The logic in line 156 is intended to make the update after the `L1InsertOrUpdate` method, where `batch_event_v2` table is updated.